tests/test_connection.py

```diff
@@ -341,3 +349,274 @@ def cb1(n):
 
     with pytest.raises(ValueError):
         conn.remove_notify_handler(cb1)
+
```

Comment:
Even if these tests are on the connection objects, I think they should be in a test module of its own. A 622 lines module, destined to increase as there are several features missing, I'd say it's too much.